compiler/rustc_session/src/config.rs

```diff
@@ -231,6 +231,37 @@ pub enum DebugInfo {
     Full,
 }
 
+/// Split debug-information is enabled by `-C split-debuginfo`, this enum is only used if split
+/// debug-information is enabled (in either `Packed` or `Unpacked` modes), and the platform
+/// uses DWARF for debug-information.
+///
+/// Some debug-information requires link-time relocation and some does not. LLVM can partition
+/// the debuginfo into sections depending on whether or not it requires link-time relocation. Split
+/// DWARF provides a mechanism which allows the linker to skip the sections which don't require
+/// link-time relocation - either by putting those sections in DWARF object files, or by keeping
+/// them in the object file in such a way that the linker will skip them.
+#[derive(Clone, Copy, Debug, PartialEq, Hash)]
+pub enum SplitDwarfKind {
+    /// Sections which do not require relocation are written into object file but ignored by the
+    /// linker.
+    Single,
```

Comment:
What is the advantage of using this when using `-Csplit-debuginfo=unpacked/packed`?

Reply:
> Looks like the motivation is primarily to retain compatability with tools that only understand object files ([source](https://reviews.llvm.org/D52296#1242164)):
> 
> > To summarise:
> > 
> > * It shares most of the benefits with the .dwo solution.
> > * Unlike .dwo, there is no need to split the file and it is friendlier to other tools (ccache, distcc, etc)
> > * But unlike .dwo a distributed build system has to copy larger .o files.

(https://github.com/rust-lang/rust/pull/89819#issuecomment-949515345)

I don't think there's any advantage to using this with `-Csplit-debuginfo=packed`, but it might be useful with `-Csplit-debuginfo=unpacked`.

Reply:
For `-Csplit-debuginfo=unpacked` we don't tell the build system where to find the debuginfo, so I don't think ccache or distcc can be safely used with `-Csplit-debuginfo=unpacked` if they were to cache dylibs or executables. For rlibs the `.dwo` files would be in the `.rlib` the build system knows about anyway, so it doesn't matter. With `-Csplit-debuginfo=unpacked` the only way to distinguish between debuginfo files for different if by assuming a specific filename scheme (and are indistinguishable if multiple crates have the same name, even if they have different `-Cmetadata`). Also with `-Csplit-debuginfo=unpacked` there would still be only a single kind of file left over after compilation. The object files are deleted after linking and the debuginfo files are kept for consumption by a debugger or other tools. Another thing is that `-Csplit-debuginfo=unpacked` produces multiple files anyway. One file for each cgu.

Reply:
Since this flag is a `-Z`, I'm comfortable with landing the implementation as is, but we should definitely revisit the concern if we ever end up considering the stabilization of the behaviour this flag exposes.